runelite-client/src/main/java/net/runelite/client/plugins/winemaking/WineMakingPlugin.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.winemaking;
+
+import com.google.common.eventbus.Subscribe;
+import java.awt.image.BufferedImage;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import static net.runelite.api.ItemID.UNFERMENTED_WINE;
+import static net.runelite.api.ItemID.UNFERMENTED_WINE_1996;
+import static net.runelite.api.ItemID.ZAMORAKS_UNFERMENTED_WINE;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDependency;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.xptracker.XpTrackerPlugin;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Wine Making"
+)
+@PluginDependency(XpTrackerPlugin.class)
+public class WineMakingPlugin extends Plugin
+{
+	private static final InventoryItemQuery UNFERMENTED_WINE_QUERY = new InventoryItemQuery(InventoryID.INVENTORY).idEquals(
+		UNFERMENTED_WINE,
+		UNFERMENTED_WINE_1996,
+		ZAMORAKS_UNFERMENTED_WINE
+	);
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private ItemManager itemManager;
+
+	private int prevWineCount;
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		infoBoxManager.removeIf(t -> t instanceof FermentationTimer);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		Item[] wines = UNFERMENTED_WINE_QUERY.result(client);
+
+		if (wines.length == prevWineCount + 1)
+		{
+			infoBoxManager.removeIf(t -> t instanceof FermentationTimer);
+
+			BufferedImage wineImage = itemManager.getImage(UNFERMENTED_WINE);
```

Comment:
You can instead call this is startup and use the .invokeLater function, that would be better than creating new image each time wine count is changed

Reply:
Is it creating a new image? I may be misunderstanding, but it looks to me like getImage() is using a cache, so most of the time it should just be doing a fairly cheap lookup and then returning the same instance. 

Everywhere else in the code, getImage() is used in this fashion.

Reply:
Hmm maybe it is fine then.